### PR TITLE
Archive pinned trash

### PR DESCRIPTION
**Pull request by mohammadkhizerkhan**
-added trash feature in Home, Archive, 
-delete and restore feature  inside trash page
*****
-pinned and unPinned feature in Home and Archive pages
*****
-archive  feature in home and archive page

### CODE REVIEW

File: src/components/notes/Notes.jsx [hunk 1]
```diff
@@ -1,54 +1,104 @@
...
-    return (
-      <>
-      {
-        notes.map(note=>{
-          const {title,desc,id}=note;
-          return(
+  const { notes } = useNote();
+  const pinnedNotes=notes.filter(item=>item.isPinned && !item.isTrash && !item.isArchive)
+  const notpinnedNotes=notes.filter(item=>!item.isPinned && !item.isTrash && !item.isArchive)
+  console.log(pinnedNotes)
```
Comment (ragini07) [inline src/components/notes/Notes.jsx]: remove console

File: src/components/notes/Notes.jsx [hunk 2]
```diff
@@ -1,54 +1,104 @@
...
 } from "react-icons/md";
-import {useNote} from "../../context/NoteContext"
-
+import { BsPin,BsPinFill } from "react-icons/bs";
+import { useNote } from "../../context/NoteContext";
+import {updateTrash,updateArchive,updatePin,updatePinnedArchive} from "../../services/firebaseServices"
 
 function Notes() {
-  const {notes} =useNote();
-  console.log(notes)
```
Comment (ragini07) [inline src/components/notes/Notes.jsx:13]: remove all consoles logs

File: src/services/firebaseServices.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import { db } from "../firebase/config";
+import { deleteDoc, doc, updateDoc } from "firebase/firestore";
+
+
+
+const updatePin=async(note)=>{
+    try {
+        await updateDoc(doc(db,"notes",note.id),{
+            isPinned:!note.isPinned,
+            // isArchive:!note.isArchive
```
Comment (ragini07) [inline src/services/firebaseServices.js]: remove comment if not needed

File: src/services/firebaseServices.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        })
+    } catch (error) {
+        console.log("error from updatePin",error)
```
Comment (ragini07) [inline src/services/firebaseServices.js:12]: suggestion : instead of console log for error you can use console error

**Review by ragini07 — COMMENTED** [review]
-Good naming convention
-Liked the UI 
-Good folder structure

File: src/services/firebaseServices.js (hunk 4 above)
Comment (mohammadkhizerkhan) [reply src/services/firebaseServices.js:12]: sure thanks

**Review by ScoobyDubeyDo — APPROVED** [review]

**Review by znjs — APPROVED** [review]

**Pull request merged**